docs/training/hyperparameter/search-methods/hp-custom.rst

```diff
@@ -0,0 +1,177 @@
+.. _topic-guides_hp-tuning-det_custom:
+
+#######################
+ Custom Search Methods
+#######################
+
++----------------------------------------------------------------+
+| API reference                                                  |
++================================================================+
+| :doc:`/reference/reference-searcher/custom-searcher-reference` |
++----------------------------------------------------------------+
+
+Determined supports defining your own hyperparameter search algorithms and provides search runner
+utilities for executing them.
+
+Remember that a Determined experiment is a set of trials, each corresponding to a point in the
+hyperparameter space.
+
+To implement a custom hyperparameter tuning algorithm, subclass
+:class:`~determined.searcher.SearchMethod`, overriding its event handler methods. For testing you
+can use :class:`~determined.searcher.LocalSearchRunner` that can be executed on your laptop. To
+achieve fault tolerance, use :class:`~determined.searcher.RemoteSearchRunner` that runs on a
+Determined cluster. and override :meth:`~determined.searcher.SearchMethod.save_method_state` and
+:meth:`~determined.searcher.SearchMethod.load_method_state` of
+:class:`~determined.searcher.SearchMethod`.
+
+You also have to specify "custom" searcher type in the experiment configuration:
+
+.. code:: yaml
+
+   searcher:
+     name: custom
+     metric: validation_loss
+     smaller_is_better: true
+     unit: batches
+
+******************************
+ Search Method Implementation
+******************************
+
+Subclass :class:`~determined.searcher.SearchMethod`. Below is a starting template:
+
+.. code:: python
+
+   import json
+   import uuid
+   from pathlib import Path
+   from typing import List
+
+   class MySearchMethod(searcher.SearchMethod):
+       def __init__(self, ...) -> None:
+           super().__init__()
+           # initialize any state you would need
+
+       def initial_operations(self) -> List[searcher.Operation]:
```

Comment:
I think we should list available Operations or provide a link to a reference page.
Copied from examples:
```
Currently, we support the following operations:
  -> Create - starts a new trial with a unique trial id and a set of hyperparameters,
  -> ValidateAfter - sets number of steps (i.e., batches or epochs) after which a validation is run for a trial
                     with a given id,
  -> Close - closes a trial with a given id,
  -> Shutdown - closes the experiment.
```